### PR TITLE
Submit button loading state

### PR DESCRIPTION
**Pull request by PixelJanitor**
kapture 2018-07-02 at 22 59 57

### CODE REVIEW

File: app/components/forms/base.js [hunk 1]
```diff
@@ -35,6 +35,9 @@ export default Component.extend({
         () => {
           this.set('isSaving', false);
           this.set('isSaveSuccessful', true);
+          setTimeout(() => {
```
Comment (cadeParade) [inline app/components/forms/base.js]: Let's set up some time to talk about why this is necessary and see if we can figure out an alternative.

**Comment by cadeParade** [comment]
@maprules1000 what do you think of this timeout addition?
Apparently we are never testing a form save click in integration tests... :/

File: app/components/forms/base.js [hunk 2]
```diff
@@ -35,6 +39,9 @@ export default Component.extend({
         () => {
           this.set('isSaving', false);
           this.set('isSaveSuccessful', true);
+          later(() => {
+            this.set('isSaveSuccessful', null);
+          }, this.get('_successIndicatorTimeout'));
```
Comment (maprules1000) [inline app/components/forms/base.js]: 👍

File: app/components/form-fields/submit.js [hunk 3]
```diff
@@ -7,7 +7,6 @@ export default Component.extend({
   disabled: false,
 
   confirmationMessage: null,
-  showSavingIndicator: true,
```
Comment (maprules1000) [inline app/components/form-fields/submit.js]: Cleaning up component implementation is a beautiful thing

File: app/templates/components/form-fields/submit.hbs [hunk 4]
```diff
@@ -1,11 +1,18 @@
-<input
+<button
   type="submit"
-  class="{{classes}}"
-  value="{{value}}"
+  class="{{classes}} relative {{if isSaving 'is-loading'}} {{if isSaveSuccessful 'is-successful'}}"
```
Comment (maprules1000) [inline app/templates/components/form-fields/submit.hbs]: It feels like there should also be a clear failure state too.

File: app/templates/components/form-fields/submit.hbs [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
-  {{saving-indicator isSaving=isSaving isSaveSuccessful=isSaveSuccessful}}
-{{/if}}
+  <span class="trans trans-opacity {{if (or isSaving isSaveSuccessful (eq isSaveSuccessful false)) 'opacity-0'}}">{{value}}</span>
+  <figure class="svg-container absolute" style="left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0)">
+    {{#if isSaving}}
+      {{inline-svg "loader-icon"}}
+    {{else if isSaveSuccessful}}
+      {{inline-svg "check-icon" class="scale-15"}}
+    {{else if (eq isSaveSuccessful false)}}
+      Failed!
```
Comment (maprules1000) [inline app/templates/components/form-fields/submit.hbs]: Yeah it looks like the current failure state is just the purple button with `Failed!` on it.

**Pull request merged**